providers/os/connection/docker/docker_file_connection.go

```diff
@@ -63,6 +63,10 @@ func NewDockerfileConnection(id uint32, conf *inventory.Config, asset *inventory
 	// this helps with running commands against the local connection
 	asset.Platform.Family = append(asset.Platform.Family, localFamily...)
 
+	if len(asset.GetConnections()) == 0 { // prevents panics by accesssing `asset.Connections[0]`
+		return nil, errors.New("no inventory connections")
+	}
+
 	if url, ok := asset.Connections[0].Options["ssh-url"]; ok {
```

Comment:
should use `conf` instead of `asset.Connections[0]`, please check if there are more places where the `asset.Connections[0]` is used

Reply:
@chris-rock I can change that, though there are some places where we populate data into `asset.Connections`, should I remove that? 🤔